pkg/controllers/capr/managesystemagent/managesystemagent_test.go

```diff
@@ -0,0 +1,74 @@
+package managesystemagent
+
+import "testing"
+
+func Test_CurrentVersionResolvesGH5551(t *testing.T) {
+	type test struct {
+		name           string
+		currentVersion string
+		expectedResult bool
+	}
+
+	tests := []test{
+		{
+			name:           "1.30 version resolved 5551",
+			currentVersion: "v1.30.4+rke2r1",
+			expectedResult: true,
+		},
+		{
+			name:           "1.30 version has not resolved 5551",
+			currentVersion: "v1.30.1+rke2r1",
+			expectedResult: false,
+		},
+		{
+			name:           "1.29 version resolved 5551",
+			currentVersion: "v1.29.8+rke2r1",
+			expectedResult: true,
+		},
+		{
+			name:           "1.29 version has not resolved 5551",
+			currentVersion: "v1.29.7+rke2r1",
+			expectedResult: false,
+		},
+		{
+			name:           "1.28 version resolved 5551",
+			currentVersion: "v1.28.13+rke2r1",
+			expectedResult: true,
+		},
+		{
+			name:           "1.28 version has not resolved 5551",
+			currentVersion: "v1.28.12+rke2r1",
+			expectedResult: false,
+		},
+		{
+			name:           "1.27 version resolved 5551",
+			currentVersion: "v1.27.16+rke2r1",
+			expectedResult: true,
+		},
+		{
+			name:           "1.27 version has not resolved 5551",
+			currentVersion: "v1.27.15+rke2r1",
+			expectedResult: false,
+		},
+		{
+			name:           "version greater than 1.30 has resolved 5551",
+			currentVersion: "v1.61.1+rke2r1",
+			expectedResult: true,
+		},
+		{
+			name:           "version less than 1.27 has not resolved 5551",
+			currentVersion: "v1.26.1+rke2r1",
+			expectedResult: false,
+		},
+	}
+	t.Parallel()
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
```suggestion
	t.Parallel()
	for _, tc := range tests {
	  tc := tc
		t.Run(tc.name, func(t *testing.T) {
```
When running in parallel, with enough tests then a scheduled test may not run before tc is changed, which will cause multiple test cases to run against the wrong value of tc. This is because although tc is a non-pointer type, in go all lambda functions capture by reference.

Reply:
Oh good to know!